modules/geo-layers/src/tile-layer/tile-layer.js

```diff
@@ -109,8 +110,11 @@ export default class TileLayer extends CompositeLayer {
     const loadingStateChanged = this.state.isLoaded !== isLoaded;
     const tilesetChanged = this.state.frameNumber !== frameNumber;
 
-    if (isLoaded && onViewportLoad && (loadingStateChanged || tilesetChanged)) {
-      onViewportLoad(tileset.selectedTiles.map(tile => tile.data));
+    if (isLoaded && (loadingStateChanged || tilesetChanged)) {
+      if (onViewportLoad) {
+        onViewportLoad(tileset.selectedTiles.map(tile => tile.data));
+      }
+      this._onViewportChange();
```

Comment:
This semantically doesn't make sense. You can call `this._onViewportLoad` here and overwrite it in the subclass to invoke `onViewportChange`. `_onViewportChange` as a base class interface will lead to future confusions because no actual viewport change is happening here.